### DISCUSSION TITLE
How to convert Vec<i64> to Vec<u64>?

### DISCUSSION THREAD

**Discussion by 3togo** [arrayfire/arrayfire-rust#338] [Q&A] [answered]
How to convert ` Vec<i64>`  to ` Vec<u64>`?

**Comment by 3togo** [comment, marked as answer]
Is there any way to avoid using for loop?
```

let input = ....
let output = Vec::<u64>::new()
for item in input.iter()
     output.push(item as u64);
```

**Reply by 9prady9** [reply]
One way to do that is by doing the following
```rust
let y = x.iter().map(|&e| e as u64).collect();
```

Also, please raise such rust generic questions in stackoverflow or rust forums. These discussions are for arrayfire rust bindings only.

**Reply by 3togo** [reply]
@9prady9 ,

many thanks

Eli
